packages/xrpl/src/models/transactions/metadata.ts

```diff
@@ -59,11 +67,35 @@ export function isDeletedNode(node: Node): node is DeletedNode {
   return Object.prototype.hasOwnProperty.call(node, `DeletedNode`)
 }
 
-export interface TransactionMetadata {
+export type TransactionMetadata<T extends BaseTransaction = Transaction> = {
   AffectedNodes: Node[]
   DeliveredAmount?: Amount
   // "unavailable" possible for transactions before 2014-01-20
   delivered_amount?: Amount | 'unavailable'
   TransactionIndex: number
   TransactionResult: string
-}
+} & (T extends Payment
```

Comment:
My only hesitation about this approach is it separates transaction specific meta from the rest of the transaction information.

Can we have each transaction file define its extended meta definition and then reference them here?  This will make it cleaner if we add more in the future.

Reply:
What names would you like to define within each transaction file?

for example, 
1.`PaymentMetadata`,`NFTokenMintMetadata`?
2.`MetadataForPayment`,`MetadataForNFTokenMint`?

Reply:
definitely the former over the latter

Reply:
Agreed.  I like `PaymentMetadata` and `NFTokenMintMetadata`.